management-api-server/src/main/java/com/datastax/mgmtapi/resources/v2/RepairResources.java

```diff
@@ -0,0 +1,118 @@
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/api/v2/repairs")
+public class RepairResources extends BaseResources {
+
+  public RepairResources(ManagementApplication application) {
+    super(application);
+  }
+
+  @PUT
+  @Operation(summary = "Initiate a new repair", operationId = "v2PutRepair")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes("application/json")
+  @ApiResponse(
+      responseCode = "200",
+      description = "Repair Successfully requested",
+      content =
+          @Content(
+              mediaType = MediaType.APPLICATION_JSON,
+              schema = @Schema(implementation = RepairRequestResponse.class),
+              examples = @ExampleObject(value = "OK")))
+  @ApiResponse(
+      responseCode = "400",
+      description = "Repair request missing Keyspace name",
+      content =
+          @Content(
+              mediaType = MediaType.TEXT_PLAIN,
+              schema = @Schema(implementation = Response.Status.class),
+              examples = @ExampleObject(value = "keyspace must be specified")))
+  @ApiResponse(
+      responseCode = "500",
+      description = "internal error, we did not receive the expected repair ID from Cassandra.",
+      content =
+          @Content(
+              mediaType = MediaType.TEXT_PLAIN,
+              schema = @Schema(implementation = Response.Status.class),
+              examples =
+                  @ExampleObject(
+                      value =
+                          "internal error, we did not receive the expected repair ID from Cassandra.")))
+  public final Response repair(RepairRequest request) {
+    return handle(
+        () -> {
+          if (request.keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity("keyspaceName must be specified")
+                .build();
+          }
+          // TODO: we need to think about how to handle unspecified/null values for the new
+          // arguments,
+          //  so that existing callers can still call v2 endpoint without args like
+          // repairParallelism etc.
+          //  I have something here but I don't like it.
+          Optional<List<RingRange>> associatedTokens =
+              request.associatedTokens == null
+                  ? Optional.empty()
+                  : Optional.of(request.associatedTokens);
+          Optional<RepairParallelism> repairParallelism =
+              request.repairParallelism == null
+                  ? Optional.empty()
+                  : Optional.of(request.repairParallelism);
+          Optional<Collection<String>> datacenters =
+              request.datacenters == null ? Optional.empty() : Optional.of(request.datacenters);
+          Optional<Integer> repairThreadCount =
+              request.repairThreadCount == null
+                  ? Optional.empty()
+                  : Optional.of(request.repairThreadCount);
```

Comment:
Thought:
Here, you do all the `null` checks so you can convert them into `Optional.empty()`. I don't see the value in cluttering this all up. The benefit of using `Optional` would be to distinguish between a missing value and an empty value. But we don't check for `null` in NodeOpsProvider, and we don't really send `null` through the RPC. So there is no difference in just sending `null` here and moving the null checks into NodeOpsProvider.

Reply:
Suggestion:
If we stay with `Optional` use here, then you can simply a lot of this
```java
          Optional<List<RingRange>> associatedTokens =
              request.associatedTokens == null
                  ? Optional.empty()
                  : Optional.of(request.associatedTokens);
```
with
```java
          Optional<List<RingRange>> associatedTokens =
              Optional.ofNullable(request.associatedTokens);
```

Reply:
I will implement this pending Olivier's feedback.